tests/test_generator.py

```diff
@@ -0,0 +1,378 @@
+#!/usr/bin/env python3
+#
+# Copyright 2021 Red Hat, Inc.
+# All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+from unittest import TestCase
+from unittest.mock import Mock
+from unittest.mock import patch
+
+import znoyder.browser
+import znoyder.downloader
+from znoyder.exclude_map import excluded_jobs_by_tag
+from znoyder.generator import GENERATED_CONFIGS_DIR
+from znoyder.generator import UPSTREAM_CONFIGS_DIR
+from znoyder.generator import additional_jobs
+from znoyder.generator import additional_jobs_by_project_and_tag
+from znoyder.generator import additional_jobs_by_tag
+from znoyder.generator import cleanup_generated_jobs_dir
+from znoyder.generator import exclude_jobs
+from znoyder.generator import excluded_jobs
+from znoyder.generator import excluded_jobs_by_project_and_tag
+from znoyder.generator import fetch_osp_projects
+from znoyder.generator import include_jobs
+from znoyder.generator import list_existing_osp_projects
+from znoyder.lib.zuul import ZuulJob
+
+
+class TestFetchOSPProjects(TestCase):
+    TEMPLATES_REPOSITORY = 'https://opendev.org/openstack/openstack-zuul-jobs'
+    PACKAGE_REPOSITORY = 'http://localhost:8080/my/package'
+
+    TEMPLATES_DIR = 'openstack/openstack-zuul-jobs/zuul.d'
+    PACKAGE_DIR = 'my/package'
+
+    def download_zuul_config(self, **kwargs):
+        match kwargs.get('repository'):
```

Comment:
Isn't the match statment new to python3.10? I think it's too soon to rely on it, our CI for example runs on py39

Reply:
I had not noticed, I will change it for some 'if...else' then.

Reply:
+1, I prefer traditional if/else, match is too new IMO

Reply:
I have uploaded a change removing the 'match' statement, this should now be compatible with Python 3.9.